Gems/Atom/RHI/Vulkan/Code/Source/RHI/Device.cpp

```diff
@@ -243,11 +244,24 @@ namespace AZ
             deviceInfo.ppEnabledExtensionNames = requiredExtensions.data();
             deviceInfo.pEnabledFeatures = &m_enabledDeviceFeatures;
 
-            const VkResult vkResult = vkCreateDevice(physicalDevice.GetNativePhysicalDevice(),
-                &deviceInfo, nullptr, &m_nativeDevice);
-            AssertSuccess(vkResult);
-            RETURN_RESULT_IF_UNSUCCESSFUL(ConvertResult(vkResult));
-
+            RHI::XRRenderingInterface* xrSystem = RHI::RHISystemInterface::Get()->GetXRSystem();
+            if (xrSystem)
+            {
+                //If a XR system is registered with RHI try to get the xr compatible Vk device from XR::Vulkan module
+                RHI::Ptr<XRDeviceDescriptor> xrDevicDescriptor = aznew XRDeviceDescriptor();
```

Comment:
Is there any reason not to just create this object on the stack?

Reply:
Vulkan::XRDeviceDescriptor inherits from RHI::XRDeviceDescriptor. Using a pointer allows me to cast as a base class an hide any information related to native data from the common XR and RHI gem